tests/test-team/pages/template-mgmt-base-page.ts

```diff
@@ -0,0 +1,51 @@
+import { Locator, type Page } from '@playwright/test';
+
+export class TemplateMgmtBasePage {
+  readonly page: Page;
+
+  readonly notifyBannerLink: Locator;
+
+  readonly loginLink: Locator;
+
+  readonly goBackLink: Locator;
+
+  readonly pageHeader: Locator;
+
+  readonly continueButton: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.notifyBannerLink = page.locator(
+      '[class="nhsuk-header__link nhsuk-header__link--service"]'
```

Comment:
What do you think about using a `data-*` attribute on HTML elements on the code instead of relying on `classnames` and `text` which are subject to change and in the case of a class return more than 1 object.

For example if we have some html:

```
<button data-test-id="pw-important-buttom">
Click me!
</button>
```

Your locators can use the data-* attribute like:

```
page.locator('data-testid=important-button')
```

I found this [article](https://playwrightsolutions.com/getbytestid/) which seems reasonably interesting and it looks like Playwright [has native support for finding elements by data-* attribute](https://playwrightsolutions.com/getbytestid/).

Reply:
This was a choice not to use the test id based on what i'm used to, and in my head, the html should be testable without the addition of special test attributes (yes, i disagree with an established test thing ... probably a losing battle). 

The fact they are subject to change makes it more valuable rather than an introduction of flakiness. While we are explicitly testing the functionality of, say a button, the use of class / role and text implicitly tests the button has the correct text and that the button is a button and not a link (both would would pass a functional test as they utilise the .click() method if using a data-testid locator that hasn't been updated).

If the element cant be found then something has gone wrong / changed by accident / unexpectedly, test id may not pick up any of this.

Happy to iterate on it / discuss if my logic sounds to picky / trying to do too much.

Reply:
Hey, that's a completely valid reason! 

You're right in saying searching by specifics like by `text` or `role` is a good way of finding elements. 

My one concern is using class names in locators as that can be quite wide ranging. More of a nitpick than anything!

Reply:
i agree with Sean here - i'm also not a fan of adding stuff to production code that's only for use by the tests